### ISSUE TITLE
Why prevent dependencies being overridden? 

### ISSUE THREAD

**Issue by rdallasgray** [young-steveo/bottlejs#36]
Hey, I was wondering why you raise an error when an existing dependency is re-registered? (See https://github.com/young-steveo/bottlejs/blob/master/src/Bottle/provider.js#L39). Our use case calls for default dependencies to be registered, but overridable to allow pluggable functionality, which I would think is a fairly common requirement.

Thanks for the project, it's been a big help to us.

**Comment by young-steveo** [comment]
@rdallasgray I hadn't considered that use-case; it seems reasonable.  Unless there's some (unlikely) problem that prevents me from doing so, I'll remove the restriction in the next minor release.

**Comment by rdallasgray** [comment]
Great. I've forked and hacked for the moment as I'm short of time, and everything seems to be working OK. If I can find the time I'll get the project set up on my machine and put together a properly-tested PR.

**Comment by young-steveo** [comment]
Cool, thanks! Though, if you don't have the time I'll probably update the library and put up a release today.

**Comment by Lukinos** [comment]
Hi, I had the same problem. As workaround, instead of registring my plugin dependencies in the main bottlejs object, I encapsulated it in another bottlejs object. For instance
```javascript
var bottle = new Bottle();
bottle.value('plugin', new Bottle());
```
Then, I recreate a bottlejs if I have to change my plugin.
```javascript
bottle.value('plugin', new Bottle());
```
In order to have access to main depenendencies in the plugin, I do a factory:
```javascript
bottle.container.plugin.factory('Beer', function(c) {
    var localBarley = c.Barley; // dependency from plugin
    var mainHops = bottle.container.Hops; // dependency from bottle
    return new Beer(localBarley, mainHops);
  });
```

**Comment by young-steveo** [comment]
I published your changes to version 1.2.0, now available in npm and bower.

**Comment by rdallasgray** [comment]
Great stuff, thanks!
